service/check_metadata.go

```diff
@@ -0,0 +1,77 @@
+package service
+import (
+	"github.com/yunify/qingstor-sdk-go/v3/request/errors"
+	"strings"
+)
+
+func IsMetaDataValid(XQSMetaData *map[string]string) error {
+	XQSMetaDataIsValid := true
+	wrongKey := ""
+	wrongValue := ""
+	if *XQSMetaData == nil {
+		XQSMetaDataIsValid = false
+	}
+	if *XQSMetaData!= nil {
```

Comment:
In golang, we always return as fast as we can.

```golang
defer func(){
	if XQSMetaDataIsValid {
		return
	}
	err = errors.ParameterValueNotAllowedError{
		ParameterName: "XQSMetaData",
		ParameterValue: "map["+wrongKey+"]="+wrongValue,
		AllowedValues: []string{"https://docs.qingcloud.com/qingstor/api/common/metadata.html"},
	}
}()

if *XQSMetaData == nil {
	XQSMetaDataIsValid = false
	return err
}
```

After that we can trim the `*XQSMetaData!= nil` branch.